### DISCUSSION TITLE
Tree / hierarchy in graph

### DISCUSSION THREAD

**Discussion by sjgknight** [kausalflow/connectome#4] [Q&A]
I think the answer to this is no, so this may be more appropriate in 'ideas', but asking on the off chance.  I wonder if there's a way to modify the graph around key nodes, particularly in a LR or TB direction.  Use cases would include (1) having RQs you're addressing, and mapping notes around these, while keeping the RQs at the top and separated from each other; (2) having multiple (overlapping) projects, with projects having special status (big funding, longevity, whatever) but many associated nodes (publications, talks, people); (3) having key themes you want to map topics around, etc.

Ideally you'd still be able to have multiple parent nodes (so, not a tree), I can't see this in the system used (I may have missed), exploring I think https://ialab.it.monash.edu/webcola/ maybe does it.

**Comment by emptymalei** [comment]
@sjgknight Thanks for posting this. I would love to improve the graph but I am not good at javascript. 
A graph that shows some kind of hierarchical structure is definitely useful. This cola.js also looks promising.

So we need to generate the data for the graph. Currently, I am using the hugo templating engine. It seems to be straightforward if the articles are arranged using the hierarchies of Hugo itself. 

Definitely worth a try.

**Comment by emptymalei** [comment]
I have some updates on this topic.

I changed the way we do graph visualization in the theme. It was an old library based on jquery. Now the theme uses G6 from AntV.

See the demo here
https://hugo-connectome.kausalflow.com/graph/

The new G6 lib is much more capable. For example, https://g6.antv.vision/en/examples/tree/dendrogram#tbDendrogram

I will explore the possibilities when I have some time.
